### PR TITLE
Add project keywords to the pyproject.toml file

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Addresses #2314 and adds keywords to pyproject.toml file.

Further suggestions are welcome!

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ readme = "README.rst"
 requires-python = ">=3.8"
 license = {text = "BSD License"}
 authors = [{name = "The PyGMT Developers", email = "pygmt.team@gmail.com"}]
-keywords = []
+keywords = ["geospatial", "cartography"]
```
Comment (yvonnefroehlich) [inline pyproject.toml:12]: Add two more keywords. Maybe we first simply collect some keywords, and determine the final order in the end 🙂?
```suggestion
keywords = ["geospatial", "cartography, geophysics, seismology"]
```

File: pyproject.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ readme = "README.rst"
 requires-python = ">=3.8"
 license = {text = "BSD License"}
 authors = [{name = "The PyGMT Developers", email = "pygmt.team@gmail.com"}]
-keywords = []
+keywords = ["cartography", "geophysics", "geospatial", "oceanography", "seismology"]
```
Comment (yvonnefroehlich) [inline pyproject.toml:12]:
```suggestion
keywords = ["cartography", "geodesy", "geophysics", "geospatial", "oceanography", "seismology"]
```

**Comment by willschlitzer** [comment]
Should "geology" be on the list of keywords or is that too generic?

**Comment by yvonnefroehlich** [comment]
> Should "geology" be on the list of keywords or is that too generic?

I am fine with "geology" being on the list of keywords. Fronts (see gallery example [Line fronts](https://www.pygmt.org/dev/gallery/lines/linefronts.html)) and the possibility to use self-created symbols can also be useful for crafting maps in geology.  Furthermore, the gallery example for the [Ternary diagram](https://www.pygmt.org/dev/gallery/basemaps/ternary.html) is also related to geology.

File: pyproject.toml [hunk 3]
```diff
@@ -9,7 +9,7 @@ readme = "README.rst"
 requires-python = ">=3.8"
 license = {text = "BSD License"}
 authors = [{name = "The PyGMT Developers", email = "pygmt.team@gmail.com"}]
-keywords = []
+keywords = ["cartography", "geodesy", "geophysics", "geospatial", "oceanography", "seismology"]
```
Comment (weiji14) [inline pyproject.toml:12]: The list is a bit long, so maybe break it into multiple lines.

```suggestion
keywords = [
    "cartography",
    "geodesy",
    "geophysics",
    "geospatial",
    "oceanography",
    "seismology"
]
```

**Review by weiji14 — APPROVED** [review]

**Review by seisman — APPROVED** [review]

**Pull request merged**